### PR TITLE
Flav/render image for model

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR enables the rendering of images from content fragments for the model. It uses GCS signed URLs to provide an image URL that providers can fetch. Currently, only OpenAI supports this method. In the coming days, we'll experiment with resizing and base64 encoding to also support Anthropic models.

Since not all models support vision, if a model is invoked in a conversation that includes content fragments with images, we will inject a placeholder instead.

### CODE REVIEW

File: front/lib/resources/content_fragment_resource.ts [hunk 1]
```diff
@@ -276,3 +284,102 @@ export async function getContentFragmentText({
...
+  try {
+    if (isSupportedImageContentFragmentType(contentType) && !excludeImages) {
+      if (!model.supportsVision) {
+        return new Ok({
+          role: "content_fragment",
+          name: `inject_${contentType}`,
+          content: [
+            {
+              type: "text",
+              text: `<attachment type="${contentType} title="${title}">[Image content interpreted by a vision-enabled model. Detailed description not available in this context.]`,
```
Comment (flvndvd) [inline front/lib/resources/content_fragment_resource.ts:328]: @spolu if you wanna double check the placeholder for model without vision support 🙏.

**Review by fontanierh — APPROVED** [review]
LFG

File: front/lib/resources/content_fragment_resource.ts (hunk 1 above)
Comment (spolu) [reply front/lib/resources/content_fragment_resource.ts:328]: Looks reasonable 👍

I would remove the "Detailed" which serves no purpose to save a token or 2 :p

File: front/lib/resources/content_fragment_resource.ts [hunk 2]
```diff
@@ -276,3 +284,102 @@ export async function getContentFragmentText({
...
+
+      return new Ok({
+        role: "content_fragment",
+        name: `inject_${contentType}`,
+        // The closing </attachment> tag will be added in the merging loop because we might
+        // need to add a "truncated..." mention in the selection loop.
+        content: [
+          {
+            type: "text",
+            text: `<attachment type="${contentType}" title="${title}">\n${content}\n`,
```
Comment (spolu) [inline front/lib/resources/content_fragment_resource.ts:367]: Not super great that this logic gets separated from the main function given that this string does not include the closing tag yet.

I would suggest fully formatting the fragment here (which will simplify token count) and then do the right thing later in the main function when we truncate it 🙏

**Review by spolu — COMMENTED** [review]
Looks great modulo 1 comment \o/

File: front/lib/resources/content_fragment_resource.ts (hunk 2 above)
Comment (flvndvd) [reply front/lib/resources/content_fragment_resource.ts:367]: Agree, but at the same time, all rendering logic should live in each message type. I can move here the token count so we properly format the content fragment directly.

File: front/lib/api/assistant/generation.ts [hunk 3]
```diff
@@ -226,30 +196,39 @@ export async function renderConversationForModelMultiActions({
...
-      for (const c of msg.content) {
+      for (const c of currentMessage.content) {
         if (!isTextContent(c)) {
           // If there is not enough room and it's an image, we simply ignore it.
           continue;
         }
 
-        const contentRes = await tokenSplit(c.text, model, remainingTokens);
+        // Remove only if it ends with "</attachment>".
+        const textWithoutClosingAttachmentTag = c.text.replace(
```
Comment (spolu) [inline front/lib/api/assistant/generation.ts:210]: surprised you didn't  create a function tokenSplitAttachment here :p

**Review by spolu — APPROVED** [review]
:100:

File: front/lib/api/assistant/generation.ts (hunk 3 above)
Comment (flvndvd) [reply front/lib/api/assistant/generation.ts:210]: Don't worry I'll take a stab at the generation and eventually create this function 😄

**Pull request merged**